demos/dotnet/dotnet/dotnet.cs

```diff
@@ -293,20 +312,39 @@ public static ProjectProperties InitializeProperties(string[] args, Log log, boo
             properties.AssemblyName = Path.GetFileNameWithoutExtension(properties.Sources[0]);
         }
 
+        string result = string.Empty;
+
+        var osName = GetOSName(Environment.OSVersion);
```

Comment:
This osName is not used anywhere atm. The tool works as discussed and as expected since OS is not taken into account (hardcoded 'win7'). I agree that it doesn't make sense to read the OS from the user's environment. I will remove the method and this function call.